src/adapters/heartcasino-polygon-1/index.ts

```diff
@@ -0,0 +1,133 @@
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+import moment from "moment";
+import BigNumber from "bignumber.js";
+import Matic from "../../sdk/matic";
+import path from "path";
+import symbolSdk from "../../sdk/symbol";
+import priceSdk from "../../sdk/price";
+
+import { ISaleEntity, ISymbolAPIResponse } from "../../sdk/Interfaces";
+import { EventData } from "web3-eth-contract";
+
+class heartcasino {
+    name: string;
+    symbol: ISymbolAPIResponse | undefined;
+    token: string;
+    protocol: string;
+    block: number;
+    contract: string;
+    events: string[];
+    pathToAbi: string;
+    range: number;
+    chunkSize: number;
+    sdk: any;
+
+    constructor() {
+        this.name = "heartcasino-polygon-1";
+        this.token = "matic";
+        this.protocol = "polygon";
+        this.block = 37411973;
+        this.contract = "0x7320D39107eF6e4Bef9513a2D4299cef9B68847A";
+        this.events = ["Transfer"];
+        this.pathToAbi = path.join(__dirname, "./abi.json");
+        this.range = 1;
+        this.chunkSize = 6;
+    }
+
+    run = async (): Promise<void> => {
+        const symbol = await symbolSdk.get(this.token, this.protocol);
+        if (!symbol) throw new Error(`Missing symbol metadata for provider ${this.name}`);
+        this.symbol = symbol;
+
+        this.sdk = await this.loadSdk();
+
+        await this.sdk.run();
+    };
+
+    loadSdk = (): any => {
+        return new Matic(this);
+    };
+
+    stop = async (): Promise<void> => {
+        this.sdk.stop();
+    };
+
+    getBuyer = async (event: EventData): Promise<string | null> => {
+        const buyer = event.returnValues.owner;
+        if (event.event === "Mint") {
+            const txReceipt = await this.sdk.getTransactionReceipt(event.transactionHash);
+            if (txReceipt === null) {
+                return null;
+            }
+            return txReceipt.from;
+        }
+
+        return buyer;
+    };
+
+    process = async (event: EventData): Promise<ISaleEntity | undefined> => {
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+        const baseTx = await this.sdk.getTransaction(event.transactionHash);
+        //if (baseTx.value == 0) {
```

Comment:
We are tracking only paid mints, sales, won auctions, etc. In short - every trade when user paid something to get NFT. Free mints are not tracked.